modules/platform/nuxeo-easyshare-core/src/main/java/org/nuxeo/easyshare/EasyShareEncryptionUtil.java

```diff
@@ -0,0 +1,180 @@
+	}
+
+	/**
+	 * Return a predefined SALT
+	 * 
+	 * @return Predefined SAL (@see SALT)
+	 */
+	public static String getDefaultSaltAsString() {
+
+		return Base64.encodeBase64String(SALT);
+	}
+
+	/**
+	 * Generate a custom salt as base64 encoded string
+	 * 
+	 * @return random base64 string
+	 */
+	public static String getNextSaltAsString() {
+
+		return Base64.encodeBase64String(getNextSaltAsByte());
+	}
+
+	/**
+	 * Encode an input string with passed salt string (default value will be
+	 * used for salt if param is empty/null) (wrapper for byte[] salt)
+	 * 
+	 * @param input
+	 *            String to encode
+	 * @param salt
+	 *            SALT to use
+	 * @return An encoded string
+	 */
+	public static String encodeWithSaltString(String input, String salt) {
+		try {
+			return encodeWithSaltByte(input, Base64.decodeBase64(salt));
+		} catch (IllegalArgumentException e) {
+			return null;
+		}
+	}
+
+	/**
+	 * Decode an encoded token with passed salt string (default value will be
+	 * used for salt if param is empty/null)
+	 * 
+	 * @param token
+	 *            Encoded String to decode using passed salt
+	 * @param salt
+	 *            SALT to use when decoding
+	 * @return Decoded string from token
+	 */
+	public static String decodeWithSaltString(String token, String salt) {
+		try {
+			return decodeWithSaltByte(token, Base64.decodeBase64(salt));
+		} catch (IllegalArgumentException e) {
+			return null;
+		}
+	}
+
+	/**
+	 * Generate a random salt
+	 * 
+	 * @return random generated salt over 8 bytes
+	 */
+	private static byte[] getNextSaltAsByte() {
+		byte[] salt = new byte[8];
+		new SecureRandom().nextBytes(salt);
+		return salt;
+	}
+
+	/**
+	 * Encode an input string with passed salt byte[] (default value will be
+	 * used for salt if param is empty/null)
+	 * 
+	 * @param input
+	 *            String to encode
+	 * @param salt
+	 *            SALT to use
+	 * @return An encoded string
+	 */
+	private static String encodeWithSaltByte(String input, byte[] salt) {
+		if (input == null) {
+			throw new IllegalArgumentException();
+		}
+		try {
+
+			byte[] randomSalt = salt != null ? salt : SALT;
+
+			KeySpec keySpec = new PBEKeySpec(null, randomSalt, ITERATION_COUNT);
+			AlgorithmParameterSpec paramSpec = new PBEParameterSpec(randomSalt, ITERATION_COUNT);
+			SecretKey key = SecretKeyFactory.getInstance("PBEWithMD5AndDES").generateSecret(keySpec);
+
+			Cipher ecipher = Cipher.getInstance(key.getAlgorithm());
+			ecipher.init(Cipher.ENCRYPT_MODE, key, paramSpec);
+
+			byte[] enc = ecipher.doFinal(input.getBytes());
+
+			String res = new String(Base64.encodeBase64(enc));
+			// escapes for url
+			res = res.replace('+', '-').replace('/', '_').replace("%", "%25").replace("\n", "%0A");
+
+			return res;
+
+		} catch (InvalidKeySpecException | NoSuchAlgorithmException | NoSuchPaddingException | InvalidKeyException
+				| InvalidAlgorithmParameterException | IllegalBlockSizeException | BadPaddingException e) {
```

Comment:
All these exceptions extends `GeneralSecurityException`, could you catch it instead?
```suggestion
		} catch (GeneralSecurityException e) {
```

Reply:
Done